patches/libuv/0001-Add-Managarm-support.patch

```diff
@@ -136,110 +194,192 @@ index 14d5f0c0..598b22c7 100644
++                       ssize_t* off_in,
++                       int fd_out,
++                       ssize_t* off_out,
++                       size_t len,
++                       unsigned int flags)
++{
++  return errno = ENOSYS, -1;
++}
++
++
++int uv__statx(int dirfd,
++              const char* path,
++              int flags,
++              unsigned int mask,
++              struct uv__statx* statxbuf) {
++  return errno = ENOSYS, -1;
++}
++
++
++ssize_t uv__getrandom(void* buf, size_t buflen, unsigned flags) {
 +  return getrandom(buf, buflen, flags);
- #else
-   return errno = ENOSYS, -1;
- #endif
++}
+diff --git a/src/unix/managarm-syscalls.h b/src/unix/managarm-syscalls.h
+new file mode 100644
+index 00000000..085afbaf
+--- /dev/null
++++ b/src/unix/managarm-syscalls.h
+@@ -0,0 +1,81 @@
++/* Copyright Joyent, Inc. and other Node contributors. All rights reserved.
++ *
++ * Permission is hereby granted, free of charge, to any person obtaining a copy
++ * of this software and associated documentation files (the "Software"), to
++ * deal in the Software without restriction, including without limitation the
++ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
++ * sell copies of the Software, and to permit persons to whom the Software is
++ * furnished to do so, subject to the following conditions:
++ *
++ * The above copyright notice and this permission notice shall be included in
++ * all copies or substantial portions of the Software.
++ *
++ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
++ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
++ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
++ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
++ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
++ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
++ * IN THE SOFTWARE.
++ */
++
++#ifndef UV_MANAGARM_SYSCALL_H_
++#define UV_MANAGARM_SYSCALL_H_
++
++#undef  _GNU_SOURCE
++#define _GNU_SOURCE
++
++#include <stdint.h>
++#include <signal.h>
++#include <sys/types.h>
++#include <sys/time.h>
++#include <sys/socket.h>
++
++struct uv__statx_timestamp {
++  int64_t tv_sec;
++  uint32_t tv_nsec;
++  int32_t unused0;
++};
++
++struct uv__statx {
++  uint32_t stx_mask;
++  uint32_t stx_blksize;
++  uint64_t stx_attributes;
++  uint32_t stx_nlink;
++  uint32_t stx_uid;
++  uint32_t stx_gid;
++  uint16_t stx_mode;
++  uint16_t unused0;
++  uint64_t stx_ino;
++  uint64_t stx_size;
++  uint64_t stx_blocks;
++  uint64_t stx_attributes_mask;
++  struct uv__statx_timestamp stx_atime;
++  struct uv__statx_timestamp stx_btime;
++  struct uv__statx_timestamp stx_ctime;
++  struct uv__statx_timestamp stx_mtime;
++  uint32_t stx_rdev_major;
++  uint32_t stx_rdev_minor;
++  uint32_t stx_dev_major;
++  uint32_t stx_dev_minor;
++  uint64_t unused1[14];
++};
```

Comment:
These structs are only ever called if `defined(__linux__)` (`src/unix/fs.c`, line 1379) and can be removed